machine/accton/accton_as_28t_ops/kernel/platform-as_28t_ops.patch

```diff
@@ -0,0 +1,479 @@
+     },
+@@ -69,28 +79,8 @@ static struct mtd_partition sflash_partition_map[] = {
+ #if defined(CONFIG_IPROC_MTD_NAND) || defined(CONFIG_IPROC_MTD_NAND_MODULE)
+ static struct mtd_partition nand_partition_map[] = {
+     {
+-        .name = "nboot",
++        .name = "open",
+         .offset = 0x00000000,
+-        .size = 2 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nenv",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 4 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nsystem",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 10 * 1024 * 1024,
+-    },
+-    {
+-        .name = "nrootfs",
+-        .offset = MTDPART_OFS_APPEND,
+-        .size = 48 * 1024 * 1024,
+-    },
+-    {
+-        .name = "ncustfs",
+-        .offset = MTDPART_OFS_APPEND,
+         .size = MTDPART_SIZ_FULL,
+     },
+ };
+diff --git a/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c b/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
+index cf16543..071c991 100755
+--- a/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
++++ b/drivers/bcmdrivers/gmac/src/et/sys/etcgmac.c
+@@ -428,6 +428,7 @@ chipattach(etc_info_t *etc, void *osh, void *regsva)
+ 	} else {
+ 		etc->phyaddr = bcm_atoi(var) & EPHY_MASK;
+ 	}
++	etc->phyaddr = 0x18;
+ 	printf("et%d: chipattach: phyaddr(0x%x)\n", etc->unit, etc->phyaddr);
+ 
+ 	/* nvram says no phy is present */
+@@ -2273,7 +2274,7 @@ chipphyrd(ch_t *ch, uint phyaddr, uint reg)
+ 	uint32 addr, bank;
+ 	uint16 tmp16;
+ 
+-	addr = phyaddr&0xf;
++	addr = phyaddr&0x1f;
+ 	bank = (phyaddr&0x1f00)>>8;
+ 
+ 	phy5221_rd_reg(ch->etc->unit, addr, bank, reg, &tmp16);
+@@ -2336,7 +2337,7 @@ chipphywr(ch_t *ch, uint phyaddr, uint reg, uint16 v)
+ #if defined(CONFIG_MACH_HR2)
+ 	uint32 addr, bank;
+ 
+-	addr = phyaddr&0xf;
++	addr = phyaddr&0x1f;
+ 	bank = (phyaddr&0x1f00)>>8;
+ 
+ 	phy5221_wr_reg(ch->etc->unit, addr, bank, reg, &v);
+diff --git a/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c b/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
+index aa8c3bd..83e7c53 100755
+--- a/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
++++ b/drivers/bcmdrivers/gmac/src/shared/nvramstubs.c
+@@ -100,6 +100,8 @@ static nvram_t fake_nvram[] = {
+ #define CONFIG_NAND_BASE		0x1c000000
+ #if (defined(CONFIG_MACH_NS) || defined(CONFIG_MACH_NSP))
+ #define CONFIG_ENV_OFFSET		0xa0000		/* 30000-b0000 - use last 10000 for env */
++#elif defined(CONFIG_MACH_HR2)
++#define CONFIG_ENV_OFFSET		0xe0000		/* e0000-effff - use 10000 for env */
+ #else
+ #define CONFIG_ENV_OFFSET		0xc0000		/* 30000-b0000 - use last 10000 for env */
+ #endif
+diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
+index 3d6beb7..f2d9f46 100644
+--- a/drivers/mtd/devices/m25p80.c
++++ b/drivers/mtd/devices/m25p80.c
+@@ -47,6 +47,15 @@
+ #define	OPCODE_CHIP_ERASE	0xc7	/* Erase whole flash chip */
+ #define	OPCODE_SE		0xd8	/* Sector erase (usually 64KiB) */
+ #define	OPCODE_RDID		0x9f	/* Read JEDEC ID */
++#define	OPCODE_RDFSR		0x70	/* Read Flag Status Register */
```

Comment:
The changes should be specific to the machine.  I am not sure whether it will affect to other platforms or not.  So that I just make them keeping in the platforms' patch.